pandas/_typing.py

```diff
@@ -122,6 +122,8 @@
 JSONSerializable = Optional[Union[PythonScalar, List, Dict]]
 Frequency = Union[str, "DateOffset"]
 Axes = Collection[Any]
+MergeTypes = Literal['inner', 'outer', 'left', 'right', 'cross']
+ConcatTypes = Literal['inner', 'outer']
```

Comment:
master currently supports python3.7 so Literal needs to be inside a TYPE_CHECKING block.

This will be changed shortly #41989, so maybe could wait for that for these changes

Reply:
Thanks for the comment; either works for me - will all other TYPE_CHECKING blocks be removed at that point? If not, it may be more consistent to just add it here as well for compatibility.

Reply:
After #41989 we can start the cleanup of the typing workarounds for Python3.7 (which we use since we don't have typing_extensions as a required dependency) so we hopefully can remove the TYPE_CHECKING that were specifically added to use Literal. (and others than were not available in the Python typing module until Python 3.8)

Reply:
I see - agreed then it makes sense to just wait until then for most of these additions. Based on Jeff's comment about waiting until 1.3, would ~2-3 weeks be a good estimate?

Reply:
yep. we've only just released the rc, so a couple of weeks at least.

Reply:
#41989 now merged.